### ISSUE TITLE
 I would like to train C3D model with my own dataset, but when i started my training loss is becoming stable after few initial iterations (after 30-40 iterations).  

### ISSUE THREAD

**Issue by sainisanjay** [facebook/C3D#250]
@dutran  I would like to train C3D model with my own data-set, but when i started my training loss is becoming stable after few initial iterations (after 30-40 iterations). I tried changing several parameters like batch size, base_lr, weight_decay and so on. but it does not change. May i know whats wrong...

**Comment by dutran** [comment]
it can happen when the problem is to simple so the training already reach minim error rate. Or the problem is too difficult and current architecture (network capacity, loss, etc) can optimize further.

**Comment by sainisanjay** [comment]
Thanks @dutran i am completely agree with your comments but if training already reached the minimum error than it should give good test accuracy. However, in my case i am getting very less test accuracy (**accuracy/top-1 = 0.062,  accuracy/top-5 = 0.246)**(as can see below after 29000 iterations).

As you mentioned that problem is too difficult: my problem seems not too difficult as i am working for Hand Gesture recognition from NVIDIA data-set. Further, i feel that my training data-set also good enough as i am using around 20204 training samples. 
```
I0426 08:30:50.227778 11301 solver.cpp:244]     Train net output #0: loss = 3.19371 (* 1 = 3.19371 loss)
I0426 08:30:50.227792 11301 sgd_solver.cpp:106] Iteration 28960, lr = 1e-11
I0426 08:31:20.966851 11301 solver.cpp:228] Iteration 28980, loss = 3.26276
I0426 08:31:20.966985 11301 solver.cpp:244]     Train net output #0: loss = 3.26276 (* 1 = 3.26276 loss)
I0426 08:31:20.966998 11301 sgd_solver.cpp:106] Iteration 28980, lr = 1e-11
I0426 08:31:50.135135 11301 solver.cpp:337] Iteration 29000, Testing net (#0)
I0426 08:33:00.506281 11301 solver.cpp:404]     Test net output #0: accuracy/top-1 = 0.062
I0426 08:33:00.506420 11301 solver.cpp:404]     Test net output #1: accuracy/top-5 = 0.246
I0426 08:33:00.506441 11301 solver.cpp:404]     Test net output #2: loss = 3.21341 (* 1 = 3.21341 loss)
I0426 08:33:00.663118 11301 solver.cpp:228] Iteration 29000, loss = 3.08495
I0426 08:33:00.663157 11301 solver.cpp:244]     Train net output #0: loss = 3.08495 (* 1 = 3.08495 loss)
I0426 08:33:00.663166 11301 sgd_solver.cpp:106] Iteration 29000, lr = 1e-11
I0426 08:33:29.043305 11301 solver.cpp:228] Iteration 29020, loss = 3.17903
I0426 08:33:29.043351 11301 solver.cpp:244]     Train net output #0: loss = 3.17903 (* 1 = 3.17903 loss)
I0426 08:33:29.043361 11301 sgd_solver.cpp:106] Iteration 29020, lr = 1e-11
I0426 08:33:59.528997 11301 solver.cpp:228] Iteration 29040, loss = 3.19435
I0426 08:33:59.529134 11301 solver.cpp:244]     Train net output #0: loss = 3.19435 (* 1 = 3.19435 loss)
I0426 08:33:59.529148 11301 sgd_solver.cpp:106] Iteration 29040, lr = 1e-11
I0426 08:34:29.590052 11301 solver.cpp:228] Iteration 29060, loss = 3.18923
I0426 08:34:29.590170 11301 solver.cpp:244]     Train net output #0: loss = 3.18923 (* 1 = 3.18923 loss)
I0426 08:34:29.590183 11301 sgd_solver.cpp:106] Iteration 29060, lr = 1e-11
I0426 08:34:59.089380 11301 solver.cpp:228] Iteration 29080, loss = 3.12322
I0426 08:34:59.089421 11301 solver.cpp:244]     Train net output #0: loss = 3.12322 (* 1 = 3.12322 loss)
I0426 08:34:59.089432 11301 sgd_solver.cpp:106] Iteration 29080, lr = 1e-11
I0426 08:35:29.156713 11301 solver.cpp:228] Iteration 29100, loss = 3.26109
I0426 08:35:29.156882 11301 solver.cpp:244]     Train net output #0: loss = 3.26109 (* 1 = 3.26109 loss)
I0426 08:35:29.156896 11301 sgd_solver.cpp:106] Iteration 29100, lr = 1e-11
I0426 08:35:59.340992 11301 solver.cpp:228] Iteration 29120, loss = 3.25801
I0426 08:35:59.341125 11301 solver.cpp:244]     Train net output #0: loss = 3.25801 (* 1 = 3.25801 loss)
I0426 08:35:59.341138 11301 sgd_solver.cpp:106] Iteration 29120, lr = 1e-11
I0426 08:36:29.662797 11301 solver.cpp:228] Iteration 29140, loss = 3.20266
I0426 08:36:29.662911 11301 solver.cpp:244]     Train net output #0: loss = 3.20266 (* 1 = 3.20266 loss)
I0426 08:36:29.662925 11301 sgd_solver.cpp:106] Iteration 29140, lr = 1e-11
I0426 08:37:00.004935 11301 solver.cpp:228] Iteration 29160, loss = 3.13623
I0426 08:37:00.005059 11301 solver.cpp:244]     Train net output #0: loss = 3.13623 (* 1 = 3.13623 loss)
I0426 08:37:00.005074 11301 sgd_solver.cpp:106] Iteration 29160, lr = 1e-11
I0426 08:37:24.117683 11301 blocking_queue.cpp:50] Data layer prefetch queue empty
I0426 08:37:30.226125 11301 solver.cpp:228] Iteration 29180, loss = 3.24377
I0426 08:37:30.226244 11301 solver.cpp:244]     Train net output #0: loss = 3.24377 (* 1 = 3.24377 loss)
I0426 08:37:30.226259 11301 sgd_solver.cpp:106] Iteration 29180, lr = 1e-11
I0426 08:38:00.557528 11301 solver.cpp:228] Iteration 29200, loss = 3.13905
I0426 08:38:00.557652 11301 solver.cpp:244]     Train net output #0: loss = 3.13905 (* 1 = 3.13905 loss)
I0426 08:38:00.557667 11301 sgd_solver.cpp:106] Iteration 29200, lr = 1e-11
I0426 08:38:30.929081 11301 solver.cpp:228] Iteration 29220, loss = 3.23426
I0426 08:38:30.929201 11301 solver.cpp:244]     Train net output #0: loss = 3.23426 (* 1 = 3.23426 loss)
I0426 08:38:30.929215 11301 sgd_solver.cpp:106] Iteration 29220, lr = 1e-11
I0426 08:39:01.289505 11301 solver.cpp:228] Iteration 29240, loss = 3.19512
I0426 08:39:01.289635 11301 solver.cpp:244]     Train net output #0: loss = 3.19512 (* 1 = 3.19512 loss)
I0426 08:39:01.289649 11301 sgd_solver.cpp:106] Iteration 29240, lr = 1e-11
I0426 08:39:31.658828 11301 solver.cpp:228] Iteration 29260, loss = 3.1813
I0426 08:39:31.658947 11301 solver.cpp:244]     Train net output #0: loss = 3.1813 (* 1 = 3.1813 loss)
I0426 08:39:31.658962 11301 sgd_solver.cpp:106] Iteration 29260, lr = 1e-11
I0426 08:40:01.915045 11301 solver.cpp:228] Iteration 29280, loss = 3.18296
I0426 08:40:01.915210 11301 solver.cpp:244]     Train net output #0: loss = 3.18296 (* 1 = 3.18296 loss)
I0426 08:40:01.915223 11301 sgd_solver.cpp:106] Iteration 29280, lr = 1e-11
I0426 08:40:31.827276 11301 solver.cpp:228] Iteration 29300, loss = 3.16723
I0426 08:40:31.827318 11301 solver.cpp:244]     Train net output #0: loss = 3.16723 (* 1 = 3.16723 loss)
I0426 08:40:31.827329 11301 sgd_solver.cpp:106] Iteration 29300, lr = 1e-11
I0426 08:41:02.031163 11301 solver.cpp:228] Iteration 29320, loss = 3.18084
I0426 08:41:02.031296 11301 solver.cpp:244]     Train net output #0: loss = 3.18084 (* 1 = 3.18084 loss)
I0426 08:41:02.031308 11301 sgd_solver.cpp:106] Iteration 29320, lr = 1e-11
I0426 08:41:32.035053 11301 solver.cpp:228] Iteration 29340, loss = 3.22927
I0426 08:41:32.035181 11301 solver.cpp:244]     Train net output #0: loss = 3.22927 (* 1 = 3.22927 loss)
I0426 08:41:32.035194 11301 sgd_solver.cpp:106] Iteration 29340, lr = 1e-11
I0426 08:42:02.131289 11301 solver.cpp:228] Iteration 29360, loss = 3.25352
I0426 08:42:02.131419 11301 solver.cpp:244]     Train net output #0: loss = 3.25352 (* 1 = 3.25352 loss)
I0426 08:42:02.131433 11301 sgd_solver.cpp:106] Iteration 29360, lr = 1e-11
I0426 08:42:32.455127 11301 solver.cpp:228] Iteration 29380, loss = 3.19554
I0426 08:42:32.455250 11301 solver.cpp:244]     Train net output #0: loss = 3.19554 (* 1 = 3.19554 loss)
I0426 08:42:32.455265 11301 sgd_solver.cpp:106] Iteration 29380, lr = 1e-11
I0426 08:43:02.782760 11301 solver.cpp:228] Iteration 29400, loss = 3.24739
I0426 08:43:02.782884 11301 solver.cpp:244]     Train net output #0: loss = 3.24739 (* 1 = 3.24739 loss)
```
**My training parameters are:** 
test_iter: 100
test_state: { stage: 'test-on-val' }
test_interval: 1000
base_lr: 0.1
momentum: 0.9
weight_decay: 0.005
gamma: 0.1
stepsize: 2693
display: 20
max_iter: 50000
snapshot: 5000
solver_mode: GPU

batch_size: 5 (i have tried with 10,20,30,40 but results was all most same)
new_height: 120
new_width: 16
new_length: 16
is_color: true
shuffle: true

**Comment by sainisanjay** [comment]
@dutran i tried as per your comment but loss is still stable. Now i am trying to compute volume mean for my images using 
GLOG_logtostderr=1 /home/sanjay/workspace/C3D/build/tools/compute_volume_mean_from_list.bin /home/sanjay/workspace/C3D/examples/c3d_nvidia/train_01_mean.lst 16 128 171 1 /home/sanjay/workspace/C3D/examples/c3d_nvidia/mean_train.binaryproto

But i am getting this error: **free() invalid next size (normal) 0x0000000000e26390**......  it seems that memory corruption error. I tried to figure out in `...../compute_volume_mean_from_list.cpp` file and `.../src/caffe/util/image_io.cpp` file but i couldn't able to.
Since you have used Mat image file from opencv which does not required memory release thus i comment img.origin.release() and img.release() but still problem occurs.

**Comment by sainisanjay** [comment]
@dutran i tried as per your comment but loss is still stable. Now i am trying to compute volume mean for my images using 
GLOG_logtostderr=1 /home/yun/Workspace/0_Gesture_recog/C3D_new/C3D-v1.0/build/tools/compute_volume_mean_from_list.bin /home/yun/Workspace/0_Gesture_recog/C3D_new/C3D-v1.0/examples/c3d_finetuning/nvidia_train_list.lst 16 128 171 1 /home/yun/Workspace/0_Gesture_recog/C3D_new/C3D-v1.0/examples/c3d_finetuning/123.binaryproto

I have make my file names as %04d.jpg thus i have changed the
`src/caffe/util/image_io.cpp` file with `sprintf(fn_im, "%s/image_%04d.jpg", img_dir, i); `

```
E0502 11:03:47.211109  6576 image_io.cpp:205] Image path: /home/yun/Workspace/1_Data/NVIDIA/Video_data/class_01/subject10_r0/duo_left/image_0001.jpg
E0502 11:03:47.211252  6576 image_io.cpp:209] Check000 
E0502 11:03:47.212024  6576 image_io.cpp:221] Check1111 
E0502 11:03:47.463919  6576 image_io.cpp:229] check data 0
E0502 11:03:47.463955  6576 image_io.cpp:246] Check333 
E0502 11:03:47.463994  6576 image_io.cpp:205] Image path: /home/yun/Workspace/1_Data/NVIDIA/Video_data/class_01/subject10_r0/duo_left/image_0002.jpg
E0502 11:03:47.464000  6576 image_io.cpp:209] Check000 
E0502 11:03:47.464264  6576 image_io.cpp:221] Check1111 
E0502 11:03:47.464350  6576 image_io.cpp:229] check data 0
*** Error in `/home/yun/Workspace/0_Gesture_recog/C3D_new/C3D-v1.0/build/tools/compute_volume_mean_from_list.bin': free(): invalid next size (normal): 0x0000000002ae8f00 ***
Aborted (core dumped)
```
